src/main/java/uk/ac/ebi/ampt2d/accession/object/ObjectAccessioningService.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.object;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import uk.ac.ebi.ampt2d.accession.AccessioningService;
+import uk.ac.ebi.ampt2d.accession.SHA1AccessionGenerator;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@ConditionalOnProperty(name = "services", havingValue = "object-accession")
+public class ObjectAccessioningService extends AccessioningService<AccessionObject, String> {
```

Comment:
What about calling this class `HashAccessioningService`? It makes more clear what it does to accession an object, and opens the door for differentiation against `MonotonicAccessioningService` or something like that.

In line with this, I think the `havingValue` should be changed to something like `hash` or `sha1`, because `object-accession` could apply both to the hash and monotonic use cases.